### PR TITLE
Adds omniscient check for AivsAi and partner

### PR DESCRIPTION
**Pull request by AlexOn1ine**
I think using battler ids here should be fine

### CODE REVIEW

File: src/battle_ai_util.c [hunk 1]
```diff
@@ -57,14 +57,15 @@ bool32 BattlerHasAi(u32 battlerId)
     switch (GetBattlerPosition(battlerId))
     {
     case B_POSITION_PLAYER_LEFT:
-        if (IsAiVsAiBattle())
+        if (AI_THINKING_STRUCT->aiFlags[battlerId] & AI_FLAG_OMNISCIENT && IsAiVsAiBattle())
```
Comment (ghoulslash) [inline src/battle_ai_util.c:60]: this is saying the player left slot has _any_ AI only if the omniscient flag is set

Comment (AlexOn1ine) [reply src/battle_ai_util.c:60]: Oh, oops.

Comment (AlexOn1ine) [reply src/battle_ai_util.c:60]: Hm, I will have to rethink that. Can I remove this specific check and keep the other one since it is still useful?

Comment (AlexOn1ine) [reply src/battle_ai_util.c:60]: @ghoulslash could you confirm that I can remove the check for now to make it ready for a merge? :eyes:

**Comment by AlexOn1ine** [comment]
#4228

**Pull request closed without merging**